packages/render/src/use-frame.tsx

```diff
@@ -377,6 +377,22 @@ export function useFrame<
 
       switch (frameButton.action) {
         case "link": {
+          // check the URL is valid
+          const locationUrl = new URL(frameButton.target);
+
+          // Reject non-http(s) URLs
+          if (
+            locationUrl.protocol !== "http:" &&
+            locationUrl.protocol !== "https:"
+          ) {
```

Comment:
Could we extract this to some sort of `isLinkTargetValid()` or something like that?